tests/components/ecobee/test_climate.py

```diff
@@ -201,11 +239,27 @@ async def test_extra_state_attributes(ecobee_fixture, thermostat):
     } == thermostat.extra_state_attributes
 
 
-async def test_is_aux_heat_on(ecobee_fixture, thermostat):
-    """Test aux heat property."""
-    assert not thermostat.is_aux_heat
-    ecobee_fixture["equipmentStatus"] = "fan, auxHeat"
-    assert thermostat.is_aux_heat
+async def test_is_aux_heat_on(hass):
+    """Test aux heat property is only enabled for auxHeatOnly."""
+    mock_get_thermostat = mock.Mock()
+    mock_get_thermostat.return_value = copy.deepcopy(
+        GENERIC_THERMOSTAT_INFO_WITH_HEATPUMP
+    )
+    mock_get_thermostat.return_value["settings"]["hvacMode"] = "auxHeatOnly"
+    with mock.patch("pyecobee.Ecobee.get_thermostat", mock_get_thermostat):
+        await setup_platform(hass, CLIMATE_DOMAIN)
+    state = hass.states.get(DEVICE_ID)
+    assert state.attributes.get("aux_heat") == "on"
+
+
+async def test_is_aux_heat_off(hass):
+    """Test aux heat property is only enabled for auxHeatOnly."""
+    mock_get_thermostat = mock.Mock()
+    mock_get_thermostat.return_value = GENERIC_THERMOSTAT_INFO_WITH_HEATPUMP
+    with mock.patch("pyecobee.Ecobee.get_thermostat", mock_get_thermostat):
+        await setup_platform(hass, CLIMATE_DOMAIN)
+    state = hass.states.get(DEVICE_ID)
+    assert state.attributes.get("aux_heat") == "off"
```

Comment:
```suggestion
    assert state.attributes["aux_heat"] == "off"
```

We should use `[]` when it would be a bug if the attribute wasn't there.

We should also use the constant instead of `"aux_heat"`